### DISCUSSION TITLE
ChebConv does not converge

### DISCUSSION THREAD

**Discussion by turnbacktime** [pyg-team/pytorch_geometric#5525] [General]
I used the features of the picture extracted from resnet50 as the input features of ChebConv, and input a randomly generated KNN graph. Why does the training loss not decrease and do not converge

**Comment by EdisonLeeeee** [comment]
Would mind providing your codes to help us figure it out?

**Reply by turnbacktime** [reply]
class ChebNet(nn.Module):
    def __init__(self, in_dim, hidden_dim, nclasses, k, dropout):
        super(ChebNet, self).__init__()
        self.conv1 = ChebConv(in_dim, hidden_dim, k, normalization='sym', bias=False)
        self.conv2 = ChebConv(hidden_dim, nclasses, k, normalization='sym', bias=False)
        self.dropout = dropout

    def reset_parameters(self):
        self.conv1.reset_parameters()
        self.conv2.reset_parameters()

    def forward(self, features, edge_index):
        x = F.relu(self.conv1(features, edge_index))
        x = F.dropout(x, p=self.dropout, training=self.training)
        x = self.conv2(x, edge_index)

        return x
----------------------------------------------------------------------------------------------------------------
The above is the two-layer ChebConv model I used. I use resnet50 to extract feature vectors from medical images, and use medical phenotype data to construct graph. I hope to classify medical images through semi supervised node classification task, but strangely, the loss in the training process has been unable to decline. When I did this on another dataset, the training was normal, but an exception occurred after the dataset was changed. I want to know what is the requirement for the feature vector input to ChebConv?

**Reply by EdisonLeeeee** [reply]
Your model looks fine. I don't think there are particular constraints for `ChebConv`. Does this phenomenon occur in other layers such as `GCNConv` as well?
